### ISSUE TITLE
[bug] Undocked Window Pane Drags Slow & Delayed

### ISSUE THREAD

**Issue by Leif-W** [notepad-plus-plus/notepad-plus-plus#5570]
### Description of the Issue
A docked window pane (such as the results of _Find All in All Opened Documents_, once undocked, when clicked and dragged by the window pane title, drags slowly & delayed, in an erratic manner.

### Steps to Reproduce the Issue
1. Create one or more documents with one or more of the same character on each line of each document
2. Open the search dialog (CTRL-F)
3. Enter the same character repeated in step one.
4. Do a _Find All in All Open Documents_.
5. The Find Result window pane will appear docked at the bottom of the window, unless it was previously undocked.
6. If the pane is docked, undock it by clicking on the title of the pane (_Find Result_), and dragging the new dockable window.
7. While dragging the window to position it in various spots, try to move in a moderate to quick speed, as if you were in a hurry and trying to get some work done.

### Expected Behavior
I expect a window that is being dragged, to respond quickly, just exactly as it I was dragging any other window around the screen, (Notepad++, any web browser such as Firefox, Waterfox, Palemoon, IE9-11, Edge, Chrome, Chromium, browser history dialogs, dockable windows in GIMP, VLC, etc, etc, etc.).  For all other GUI applications I have ever used since C64/GEOS or WfW 3.10, through most recent versions of Windows, (or any app in any other OS such as Mac, FreeBSD if installed with a windowing environment, any Linux distro), this is a non-issue.  I expect a draggable window to behave in the well-defined manner that draggable windows have behaved in for over 3 decades.  Basic UX issues.

### Actual Behavior
Notice how the window drags slowly, such that the movements are delayed behind current mouse position.  This delay makes it difficult to predict final position of window as we can not see the actual current position of the mouse or the outline of the dialog box.  This bug makes precision placement of the window difficult and very tedious and time consuming.

### Debug Information
Closing the issue (#1809) does not simply make it not exist.  I am already on 7.6.6.  I update regularly.  Issues exists with no plugins (closed Notepad++, moved plugins files to external folder like C:\temp so Notepad++ can not find them, restart Notepad++, behavior unchanged, unaffected by installed plugins).  Prematurely closing bugs doesn't make things cleaner, it just adds more clutter, as the same problem will be described in various ways, multiple times.

**Comment by Uhf7** [comment]
I can reproduce the same or a very similar problem with the current Notepad++ version 7.9 under Windows 10 1607. Can anyone confirm this behavior?

### Steps to Reproduce the Issue
- Unpack a `npp.7.9.portable.zip` into a new directory.
- Close all running applications on your computer, except File Explorer.
- Use File Explorer to start the newly unpacked Notepad++.exe.
- Display a Project Panel 1 (View -> Project Panels -> Project Panel 1).
- Make the Project Panel float by double-clicking it's header.
- Drag the Project Panel around a little to see, that everything works smoothly.
- Then, start a Windows UWP app. I tested
  - Alarms & Clock
  - Calculator
  - Calendar
  - the "classic" MS Edge Browser (The current MS Edge is a Win32 application).
- As soon as any Windows UWP app is running in non-minimized state together with Notepad++, then dragging around the Project Panel isn't smooth anymore.

### Debug Information
Notepad++ v7.9   (32-bit)
Build time : Sep 22 2020 - 03:24:22
Path : C:\Tools\Npp_79\notepad++.exe
Admin mode : OFF
Local Conf mode : ON
OS Name : Windows 10 Pro (64-bit) 
OS Version : 1607
OS Build : 14393.576
Current ANSI codepage : 1252
Plugins : mimeTools.dll NppConverter.dll NppExport.dll

**Comment by ArkadiuszMichalski** [comment]
@Uhf7 But without Windows UWP app your undocked Project Panel works smoothly (like other windows)? Or move some slowly, but with Windows UWP app works even worse?

@Leif-W It's your second bug and you still didn't include debug information

**Comment by Uhf7** [comment]
It works correctly without a not minimized UWP app in the background, and turns bad, as soon as a not minimized UWP app runs in the background. Usually the bad effect appears on my computer, when I search something in the classic MS Edge (UWP) browser and then switch back to Notepad++ with Alt-Tab, the browser still open in the background.

**Comment by ArkadiuszMichalski** [comment]
I can't reproduce for:
```
Notepad++ v7.9   (32-bit)
Build time : Sep 22 2020 - 03:24:22
Path : D:\Programy\NPP\notepad++.exe
Admin mode : OFF
Local Conf mode : ON
OS Name : Windows 10 Home (64-bit) 
OS Version : 2004
OS Build : 19041.508
Current ANSI codepage : 1250
Plugins : mimeTools.dll NppConverter.dll NppExport.dll 
```
But I see that in Win10 this undoced panels works faster than on Win7, but still not same as other standard windows.

**Comment by Uhf7** [comment]
@ArkadiuszMichalski,

Thank you for testing, perhaps this behavior is improved in Windows 10 2004 systems. On my system, it is clearly bipolar, smooth without UWP apps, and about 5 seconds stall after each drag with UWP apps. Within this 5 seconds, I cannot initiate anything else (even switch to another application is impossible), and the side window which has been dragged isn't updated.

I can get rid of the stalls by removing the mouse_event code line from Gripper.cpp

https://github.com/notepad-plus-plus/notepad-plus-plus/blob/a1031517742c7ef600b825bf9f193b3f2740dff6/PowerEditor/src/WinControls/DockingWnd/Gripper.cpp#L240)

And everything still works fine after removing the mouse hook function from the same source file, the only thing I don't know, what this is/was for in the first place.

**Comment by ArkadiuszMichalski** [comment]
In fact, 5s is a noticeable difference, maybe other Win10 users would add their comments here, which would be helpful. From my perspective this slower undocked window dragging is not a problem (ion both systems), but I don't have that much lag.

**Comment by Ekopalypse** [comment]
I have a similar/same issue on but know that this issue happened to start some time ago. Maybe even before 7.5

Notepad++ v7.9   (64-bit)
Build time : Sep 22 2020 - 03:19:04
Path : D:\...\notepad++.exe
Admin mode : OFF
Local Conf mode : ON
OS Name : Windows 7 Professional (64-bit) 
OS Build : 7601.24546
Current ANSI codepage : 1252
Plugins : ComparePlugin.dll Explorer.dll mimeTools.dll NppConverter.dll NppExport.dll PythonScript.dll XMLTools.dll 

Here a gif which, hopefully, shows the slowness by concentrating on the cursor and where it ends after a move of the panel.

![slow_move](https://user-images.githubusercontent.com/47723516/95768024-f7be7480-0cb5-11eb-888a-0f6d24b375c2.gif)

**Comment by Uhf7** [comment]
@Ekopalypse,

thank you for the gif, it is an effect very similar to what I see.

The same is, that a kind of blockade occurs after dragging the window to another place, and that the next dragging is only possible, after this blockade ends.

Different is, that on your system the side window content is updated correctly after the move, while at my Windows 10 system, the content gets updated after the "blockade". While it is blocking, I see an empty or partly updated window.

And, I can't reproduce it on my Windows 7:

	Notepad++ v7.9   (32-bit)
	Build time : Sep 22 2020 - 03:24:22
	Path : C:\Tools\Npp\notepad++.exe
	Admin mode : OFF
	Local Conf mode : ON
	OS Name : Windows 7 Professional (64-bit) 
	OS Build : 7601.0
	Current ANSI codepage : 1252
	Plugins : DSpellCheck.dll LuaScript.dll mimeTools.dll NppConverter.dll NppExport.dll

**Comment by Ekopalypse** [comment]
@Uhf7, yeah, I have a laptop with the same version of Windows and the same Notepad++ setup and it does not appear there.
Not sure what is going on here.

**Comment by ArkadiuszMichalski** [comment]
@Ekopalypse What program are you using to record animated gifs? I check ScreenToGif bad get some artifacts when dragging panels. Btw, in my setups after start dragging my mouse cursor never goes beyond the panel bar (even if it moves slowly), it always remains where I clicked left mouse button (relative to panel bar).

**Comment by Ekopalypse** [comment]
@ArkadiuszMichalski - I use ScreenToGif as well. Version 2.13.3 with default settings.

**Comment by KnIfER** [comment]
Drawing big rectangles  on the desktop HDC is slow. Drawing on the main window's HDC when it's maximized is smoother but has glitches.

> Refer to Gripper.cpp

The best solutions is —— Move a transparent window instead.

After days of tweaking and debugging, I now have this : 

![20210328_012714_SchivoNo 0_GifNo 0](https://user-images.githubusercontent.com/11593903/112727256-17481180-8f65-11eb-8750-09ed539e65fc.gif)

Tweaking this new docking preview system with Windows' Aero-Snapping is so hard, but eventually it works.

**Comment by KnIfER** [comment]
I introduced a variable `DockingPreviewMethod `, so that you can see  the difference ：

### DockingPreviewMethod =0
Draw on the Desktop HDC

### DockingPreviewMethod =1
Move a Transparent window
### DockingPreviewMethod =2
For docked containers : Move a Transparent window. 

For floating containers : Move the window directly. While moving, if the cursor position hit some other dockable container, display the  transparent window as a preview.
